libs/tree/llrb.go

```diff
@@ -85,28 +79,16 @@ func (t *llrb) GetNext(starter *NodeKey, predicate func(interface{}) bool) (inte
 	return candidate.data, candidate.key, nil
 }
 
-func (t *llrb) UpdateKey(oldKey NodeKey, newKey NodeKey) error {
-	data, err := t.Remove(oldKey)
-	if err != nil {
-		return err
-	}
-	return t.Insert(newKey, data)
-}
-
 // Insert inserts value into the tree
-func (t *llrb) Insert(key NodeKey, data interface{}) error {
+func (t *llrb) Insert(key NodeKey, data interface{}) {
```

Comment:
is line 105 still relevant?

Reply:
Yes, the unexported insert need this error to decide whether the size needs to increase, this error will not be passed outside

Reply:
we should at least make it private if it will never be used outside. also, can we early return at line 105?